packages/bot-strategy-runner/src/ConfigBuilder.ts

```diff
@@ -0,0 +1,203 @@
+// Config builder that constructs whitelists and bot configs. Used to parameterize the strategy runner.
+import Web3 from "web3";
+const { toChecksumAddress } = Web3.utils;
+
+import { getWeb3 } from "@uma/common";
+import { getAbi } from "@uma/core";
+
+import { liquidatorConfig, disputerConfig, monitorConfig } from "./BotEntryWrapper";
+import nodeFetch from "node-fetch";
+import assert from "assert";
+
+const supportedBotTypes = ["liquidator", "disputer", "monitor"];
+
+export interface botSettings {
+  enableBotType: boolean;
+  addressWhitelist?: Array<string>;
+  addressBlacklist?: Array<string>;
+  commonConfig?: { [key: string]: any };
+  addressConfigOverride?: { [key: string]: { [key: string]: any } };
+}
+
+export interface strategyRunnerConfig {
+  botNetwork?: string;
+  pollingDelay?: number;
+  botConcurrency?: number;
+  strategyTimeout?: number;
+  verboseLogs?: boolean;
+  emitDebugLogs?: boolean;
+  globalAddressWhitelistUrls?: Array<string>;
+  globalAddressWhitelist?: Array<string>;
+  globalAddressBlacklist?: Array<string>;
+  commonConfig?: { [key: string]: any };
+  liquidatorSettings?: botSettings;
+  disputerSettings?: botSettings;
+  monitorSettings?: botSettings;
+}
+
+// The global whitelist is the concatenation of the any `globalAddressWhitelist` plus any values stored on remote json
+// files define in `globalAddressWhitelistUrls`.
+export async function buildGlobalWhitelist(config: strategyRunnerConfig) {
+  let whitelist = config.globalAddressWhitelist ? config.globalAddressWhitelist : [];
+
+  if (config.globalAddressWhitelistUrls) {
+    const responses = await Promise.all(config.globalAddressWhitelistUrls.map((url: string) => nodeFetch(url)));
+    const responseJson = await Promise.all(responses.map((response: any) => response.json()));
+    responseJson.forEach((contractAddressesResponse: any) => {
+      if (contractAddressesResponse.empWhitelist) whitelist = [...whitelist, ...contractAddressesResponse.empWhitelist];
+      else console.log("Global Whitelist file does not have the `empWhitelist` key or is malformed");
```

Comment:
why not throw error?

Reply:
ye, you right. will throw an error here.

Reply:
well actually there was a reason for this. it is possible that the global whitelist is malformed but everything else is still fine. in that case the way it is structure here the bot will continue to work as expected, minus the ones from the whitelist.

However, this is pretty difficult to reason about so I think we should ideally throw an error like you prescribed. have updated accordingly.